Mathlib/Algebra/Order/Hom/Ultra.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2024 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Algebra.Order.Hom.Normed
+import Mathlib.Topology.MetricSpace.Ultra.Basic
+
+/-!
+# Constructing nonarchimedean (ultrametric) normed groups from nonarchimedean normed homs
+
+This file defines constructions that upgrade `(Comm)Group` to `(Semi)Normed(Comm)Group`
+using a `Group(Semi)normClass` when the codomain is the reals and the hom is nonarchimedean.
+
+## Implementation details
+
+The lemmas need to assume `[Dist α]` to be able to be stated, so they take an extra argument
+that shows that this distance instance is propositionally equal to the one that comes from the
+hom-based `AddGroupSeminormClass.toSeminormedAddGroup f` construction. To help at use site,
+the argument is an autoparam that resolves by definitional equality when using these constructions.
+-/
+
+variable {F α : Type*} [FunLike F α ℝ]
+
+/-- Proves that an `AddSeminormedGroup` structure constructed from an `AddGroupSeminormClass`
+that is `IsNonarchimedean` satisfies `IsUltrametricDist`. -/
```

Comment:
```suggestion
/-- Proves that an `AddSeminormedGroup` structure constructed from an `AddGroupSeminormClass`
that `IsNonarchimedean` satisfies `IsUltrametricDist`. -/
```
Maybe?

Reply:
the seminorm is nonarchimedean. i'll rephrase